src/llvm-cpufeatures.cpp

```diff
@@ -108,24 +122,54 @@ bool lowerCPUFeatures(Module &M)
     }
 }
 
-struct CPUFeatures : PassInfoMixin<CPUFeatures> {
-    PreservedAnalyses run(Module &M, ModuleAnalysisManager &AM);
+struct CPUFeaturesPass : public PassInfoMixin<CPUFeaturesPass> {
+  static void registerCallbacks(PassBuilder &PB) {
+    PB.registerPipelineParsingCallback(
+        [](StringRef Name, ModulePassManager &PM,
+           ArrayRef<PassBuilder::PipelineElement> InnerPipeline) {
+          if (Name == "CPUFeatures") {
+            PM.addPass(CPUFeaturesPass());
+            return true;
+          }
+          return false;
+        });
+  }
+
+  PreservedAnalyses run(Module &M, ModuleAnalysisManager &AM);
 };
 
-PreservedAnalyses CPUFeatures::run(Module &M, ModuleAnalysisManager &AM)
+
+PreservedAnalyses CPUFeaturesPass::run(Module &M, ModuleAnalysisManager &AM)
 {
-    lowerCPUFeatures(M);
+    auto &MMI = AM.getResult<MachineModuleAnalysis>(M);
+    auto &TM = MMI.getTarget();
+    lowerCPUFeatures(TM, M);
     return PreservedAnalyses::all();
 }
 
+extern "C" JL_DLLEXPORT ::llvm::PassPluginLibraryInfo
+llvmGetPassPluginInfo() {
```

Comment:
We should move that out to a different file, since we will need to add all the other NewPM to this as well.

Reply:
You have all the other NewPM changes, right? So I will leave this up to you to move.

Reply:
There are already other passes that support the NewPM, but sure I can do that.